### PR TITLE
Changed namespace handling in mock support to be explicit

### PR DESCRIPTION
**Pull request by andy-maier**
The commit message details all changes, explains the background, and should be read before reviewing.
Ready for review and merge.

### CODE REVIEW

File: testsuite/test_wbemconnection_mock.py [hunk 1]
```diff
@@ -1559,11 +1667,12 @@ def test_getclass(self, conn, tst_class):
...
         "ns, cln, tst_ns, exp_er", [
-            [None, 'CIM_Foo', None, None],
-            [None, 'cim_foo', None, None],
-            [None, 'CIM_Foo', 'root/blah', CIM_ERR_INVALID_NAMESPACE],
-            ['root/blah', 'CIM_Foo', 'root/blah', None],
-            [None, 'CIM_Foox', None, CIM_ERR_NOT_FOUND],
+            [DEFAULT_NAMESPACE, 'CIM_Foo', None, None],
+            [DEFAULT_NAMESPACE, 'cim_foo', None, None],
+            [DEFAULT_NAMESPACE, 'CIM_Foo', 'BadNamespace',
+             CIM_ERR_INVALID_NAMESPACE],
```
Comment (KSchopmeyer) [inline testsuite/test_wbemconnection_mock.py]: Doesn't this actually change the test? Most of the tests were None and one was a namespace?  This has namespace DEFAULT_NAMESPACE for all of the tests. I have not looked elsewhere to see if the same question existsyet.

File: testsuite/test_wbemconnection_mock.py [hunk 2]
```diff
@@ -2161,12 +2270,15 @@ class TestInstanceOperations(object):
 
     @pytest.mark.parametrize(
         "ns, cln, inst_id, tst_ns, exp_er", [
-            [None, 'CIM_Foo', 'CIM_Foo1', None, None],
-            ['root/blah', 'CIM_Foo', 'CIM_Foo1', 'root/blah', None],
-            ['blah', 'CIM_Foo', 'badid', 'blah', CIM_ERR_NOT_FOUND],
-            ['blah', 'CIM_Foo', 'CIM_Foo1', 'whoop',
+            [DEFAULT_NAMESPACE, 'CIM_Foo', 'CIM_Foo1', None, None],
```
Comment (KSchopmeyer) [inline testsuite/test_wbemconnection_mock.py]: Original test was None for ns for this test.

File: pywbem_mock/_wbemconnection_mock.py [hunk 3]
```diff
@@ -356,14 +378,91 @@ def __repr__(self):
     #
     ################################################################
 
+    def add_namespace(self, namespace):
+        """
+        Add a CIM namespace to the mock repository.
+
+        The namespace must not yet exist in the mock repository.
+
+        Note that the default connection namespace automatically exists in the
```
Comment (KSchopmeyer) [inline pywbem_mock/_wbemconnection_mock.py]: This comment may require some more explanation.  I had to think about it for a moment.  Maybe something like "It is created when the _wbemconnection_mock is constructed."  Also might better refernce with conn.default_namespace.

Question. What happens if you do delete_namespace on the default conneciton namespace

File: testsuite/test_wbemconnection_mock.py (hunk 1 above)
Comment (andy-maier) [reply testsuite/test_wbemconnection_mock.py]: The `ns` parameter is used as the namespace in which the CIM objects are set up. The `tst_ns` parameter is the namespace against which the operation to be tested (GetClass in this case) is run. It does not make sense to set up CIM objects during test preparation n a namespace `None`, if that is not the purpose of the test.

File: testsuite/test_wbemconnection_mock.py (hunk 2 above)
Comment (andy-maier) [reply testsuite/test_wbemconnection_mock.py]: Same response as above.

File: pywbem_mock/_wbemconnection_mock.py (hunk 3 above)
Comment (andy-maier) [reply pywbem_mock/_wbemconnection_mock.py]: On  the comment: Changed that to: "Note that the default connection namespace is automatically added to the mock repository upon creation of this object."

On deleting the default connection namespace: There is no externally available `delete_namespace()` method on the `FakedWBEMConnection` class (only an internal `_remove_namespace()`). However, the faked DeleteInstance operation would remove it. I have added a check in the internal `_remove_namespace()` method that raises CIMError(CIM_ERR_NAMESPACE_NOT_EMPTY) if the default connection namespace is attempted to be removed, and a description of that behavior in the faked DeleteInstance operation.

**Comment by andy-maier** [comment]
Comments resolved; some have been rejected, some addressed by an update.

**Pull request merged**